src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/FSharpCoreReflectionProxy.cs

```diff
@@ -0,0 +1,224 @@
+            _fsharpSetCtor = fsharpCoreAssembly.GetType("Microsoft.FSharp.Collections.SetModule")?.GetMethod("OfSeq", BindingFlags.Public | BindingFlags.Static);
+            _fsharpMapCtor = fsharpCoreAssembly.GetType("Microsoft.FSharp.Collections.MapModule")?.GetMethod("OfSeq", BindingFlags.Public | BindingFlags.Static);
+        }
+
+        [RequiresUnreferencedCode(FSharpCoreUnreferencedCodeMessage)]
+        public FSharpKind DetectFSharpKind(Type type)
+        {
+            Attribute? compilationMappingAttribute = GetFSharpCompilationMappingAttribute(type);
+
+            if (compilationMappingAttribute is null)
+            {
+                return FSharpKind.Unrecognized;
+            }
+
+            if (type.IsGenericType)
+            {
+                Type genericType = type.GetGenericTypeDefinition();
+                if (genericType == _fsharpOptionType) return FSharpKind.Option;
+                if (genericType == _fsharpListType) return FSharpKind.List;
+                if (genericType == _fsharpSetType) return FSharpKind.Set;
+                if (genericType == _fsharpMapType) return FSharpKind.Map;
+            }
+
+            return (GetSourceConstructFlags(compilationMappingAttribute) & SourceConstructFlags.KindMask) switch
+            {
+                SourceConstructFlags.RecordType => FSharpKind.Record,
+                _ => FSharpKind.Unrecognized
+            };
+        }
+
+        [RequiresUnreferencedCode(FSharpCoreUnreferencedCodeMessage)]
+        public Func<TFSharpOption, T> CreateFSharpOptionValueGetter<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicMethods)] TFSharpOption, T>()
+        {
+            Debug.Assert(typeof(TFSharpOption).GetGenericTypeDefinition() == _fsharpOptionType!);
+            MethodInfo valueGetter = EnsureMemberExists(typeof(TFSharpOption).GetMethod("get_Value", BindingFlags.Public | BindingFlags.Instance), "Option.Value");
+            return CreateDelegate<Func<TFSharpOption, T>>(valueGetter);
+        }
+
+        [RequiresUnreferencedCode(FSharpCoreUnreferencedCodeMessage)]
+        public Func<T, TFSharpOption> CreateFSharpOptionConstructor<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicMethods)] TFSharpOption, T>()
+        {
+            Debug.Assert(typeof(TFSharpOption).GetGenericTypeDefinition() == _fsharpOptionType!);
+            MethodInfo methodInfo = EnsureMemberExists(typeof(TFSharpOption).GetMethod("Some", BindingFlags.Public | BindingFlags.Static), "Option.Some");
+            return CreateDelegate<Func<T, TFSharpOption>>(methodInfo);
+        }
+
+        [RequiresUnreferencedCode(FSharpCoreUnreferencedCodeMessage)]
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2060:MakeGenericMethod", Justification = "The public methods are marked RequiresUnreferencedCode.")]
```

Comment:
You don't need both `RequiresUnreferencedCode` and `UnconditionalSuppressMessage` attributes on these methods. You can just apply `RequiresUnreferencedCode`, and that will suppress any warning from inside this method, and propagate the warning to any callers.